### ISSUE TITLE
Velocity 1.20.2: Trials and Tribulations of Cross-Server Chat

### ISSUE THREAD

**Issue by JerichoTorrent** [kangarko/ChatControl#2509]
### "/version ChatControlRed" - plugin version

10.22.5

### Are you using MySQL?

Yes

### "/chc debug" output (strongly recommended)

[debug.zip](https://github.com/kangarko/ChatControl-Red/files/13710831/debug.zip)

### Information about the issue/bug

I use VelocityControl on the proxy (Velocity 1.20.2) and ChatControl Red 10.22.5 on my 3 1.20.2 backend servers. I provided a debug from one backend. My issue is twofold.
1. I can't find either in the wiki, discord, or through my trial and error a tried and true method to enable cross-server chat in my global channel. I have my settings.yml channels list written as follows:
```
  List:
    Survival:
      Format: chat
      Format_Spy: '@noprefix chat'
      Discord_Channel_Id: 1100905925507817533
      Format_Discord: none
    Global:
      Format: global-chat
      Discord_Channel_Id: 1182819310574518393
      Bungee: true
    Admin:
      Format: admin-chat
      Format_Console: '[Admin-chat {player}] {message}'
      Discord_Channel_Id: 1100917607642693672
      Bungee: true
      Format_Discord: none
    Ranged:
      Format: '&6[ranged] &7{pl_prefix+}{player}:{pl_suffix} {chat_color}{chat_decoration}{message}'
      Range: 100
      Bungee: false
    Staff:
      Format: staff
      Bungee: true
```
All 3 backends are formatted very similarly. The only differences are the "survival" channel is of course named differently on other servers and has different discord channel IDs. That's just context and not relevant. I also have the bungee.integration set to true. I assume the word "Bungee" can be substituted for Velocity, since I'm not seeing an official full-suite integration of Velocity into this plugin. No matter what I try I can't read messages from other servers even though I am in the global channel. Same thing for staff and admin channels.
2. I use VelocityControl as mentioned prior and I'm running into a couple issues. I have join and quit messages turned off on the backend in favor of VelocityControl's join, quit and switch messages which I like much more. However, sometimes VelocityControl misses a player joining. I'll see people chat without me seeing them having ever joined. I'll see people leave then see them still on the server, because VelocityControl missed when they joined back. Is there any help you can provide on how to set this up properly? It just seems like a lack of consistency with catching join and quit messages. Before it is asked, yes the server names in Velocity and VelocityControl are setup right.
Any clarity you can provide on both issues would be most appreciated.

**Comment by james090500** [comment]
Please make sure BungeeCord is enabled at the very bottom of settings.yml 

![image](https://github.com/kangarko/ChatControl-Red/assets/20372645/0a907b91-0209-4ce9-95df-05f633647f80)

Regarding velocity join messages I am aware and tryign to track it down

**Comment by JerichoTorrent** [comment]
> Please make sure BungeeCord is enabled at the very bottom of settings.yml
> 
> ![image](https://private-user-images.githubusercontent.com/20372645/291692809-0a907b91-0209-4ce9-95df-05f633647f80.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMwMTk2MTAsIm5iZiI6MTcwMzAxOTMxMCwicGF0aCI6Ii8yMDM3MjY0NS8yOTE2OTI4MDktMGE5MDdiOTEtMDIwOS00Y2U5LTk1ZGYtMDVmNjMzNjQ3ZjgwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDIwNTUxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3OThhZTAyYzQxOGNhYTcxMTVjMDY4NzQ3YzljOWY4NzQzMzViMWQxZjBhZWIxNDgyNDNlYmMyMjMyMmU0NWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aEKiw5d2EXsTylqgY-muEDmNvggbVkeT8y673LjVP2Q)
> 
> Regarding velocity join messages I am aware and tryign to track it down

Thanks, yes BungeeCord is enabled. I will be patient if a fix has to be implemented, but if there’s anything I can do to get messages across backend servers to appear in the global chat channel I setup, please let me know. I can only see messages in the server I am currently playing in.

**Comment by kangarko** [comment]
Hey!

Can you try writing Bungee option with the value on true inside settings.yml where you have your global channel?

![Screenshot 2023-12-23 at 18 55 57](https://github.com/kangarko/ChatControl-Red/assets/1918426/3b542821-19ec-4cdb-b063-0d5b72900d9e)

I just tested, got it working even without that option but just in case. If that won't help, set Debug key in settings.yml on spigot and velocity to "[bungee]" and you should see some logs that might help.

And if even these won't, please ping @james090500 as this might be velocity related.

Matej

**Comment by kangarko** [comment]
Closing due to inactivity, comment below if you still need help?

**Comment by JerichoTorrent** [comment]
@kangarko @james090500 thank you both for your help. I have been very sick so have taken a step back from server development, sorry about that.
I have no idea what I did, perhaps some combination of adding or removing the bungee integration and bungee: true from the list, to maybe a couple other changes but now it works perfectly. Wish I was able to provide a fix to others with the same issue but I’m assuming just doing the above as intended should work.
As for VelocityControl missing join messages, looks like James was looking into that. Not really a big deal to me, but wish you luck if it’s something that requires a fix and it’s not on my end. Thanks again for the help.
